sandy/decay.py

```diff
@@ -148,6 +157,78 @@ def get_decayconstant(self, with_uncertainty=True):
         else:
             return df.LAMBDA
 
+    def custom_perturbation_decayconstant(self, pert, zam):
+        """
+        Apply a custom perturbation to the decay constant for a given ZAM.
+
+        Parameters
+        ----------
+        pert : `float`
+            Perturbation coefficient as ratio value.
+        zam : `int`
+            ZAM number of the material to which perturbation is to be
+            applied.
+
+        Returns
+        -------
+        `sandy.decay.DecayData`
+            DacayData instance with given ZAM and decay constant perturbed.
+
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file("jeff_33", 'decay', [922350, 922380])
+        >>> rdd = sandy.DecayData.from_endf6(endf6)
+        >>> dc_pert = rdd.custom_perturbation_decayconstant(0.1, 922350)
+        >>> dc_pert.data
+        {922350: {'half_life': 2.22102e+16,
+          'decay_constant': 3.432935762018982e-17,
+          'decay_constant_uncertainty': 2.2171470275223715e-20,
+          'stable': False,
+          'spin': 3.5,
+          'parity': -1.0,
+          'decay_energy': {'beta': 50671.7,
+           'gamma': 163616.0,
+           'alpha': 4464600.0,
+           'total': 4678887.7},
+          'decay_energy_uncertainties': {'beta': 4291.63,
+           'gamma': 1708.01,
+           'alpha': 163255.0,
+           'total': 163320.33067324167},
+          'decay_modes': [('40x0',
+            {'decay_products': {902310: 1.0, 20040: 1.0},
+             'branching_ratio': 1.0,
+             'branching_ratio_uncertainty': 0.0001}),
+           ('60x0',
+            {'decay_products': {},
+             'branching_ratio': 7.2e-11,
+             'branching_ratio_uncertainty': 2.1e-11})]},
+         922380: {'half_life': 1.40996e+17,
+          'decay_constant': 4.916076913954618e-18,
+          'decay_constant_uncertainty': 3.3008662253228094e-21,
+          'stable': False,
+          'spin': 0.0,
+          'parity': 1.0,
+          'decay_energy': {'beta': 10208.3,
+           'gamma': 1100.17,
+           'alpha': 4258770.0,
+           'total': 4270078.47},
+          'decay_energy_uncertainties': {'beta': 851.158,
+           'gamma': 104.433,
+           'alpha': 29639.0,
+           'total': 29651.4029548764},
+          'decay_modes': [('40x0',
+            {'decay_products': {902340: 1.0, 20040: 1.0},
+             'branching_ratio': 0.999999,
+             'branching_ratio_uncertainty': 1e-08}),
+           ('60x0',
+            {'decay_products': {},
+             'branching_ratio': 5.46e-07,
+             'branching_ratio_uncertainty': 1e-08})]}}
+        """
+        pert_dc = copy.deepcopy(self.data)
+        pert_dc[zam]['decay_constant'] = pert_dc[zam]['decay_constant'] * (1 + pert)

```

Comment:
We should use perturbation coefficients as ratios to be consistent with the xs part.
Example, pert=1.05 for a 5% perturbation.
can you adjust it? also the docstring...